collectors/30/kafka_offset_collector.py

```diff
@@ -40,60 +45,78 @@ def format_kafka_consumer_offset_tsd_key(consumer_group, topic, partition, offse
         offset = offset_response.offsets[0]
         total_offset += offset
-        print format_kafka_broker_offset_tsd_key(topic, partition, offset)
+        print format_kafka_broker_offset_tsd_key(env, topic, partition, offset)
     return total_offset
 
 
-def report():
-    # see the monitoring.json chef role
-    zk_quorums = os.getenv('MONITORED_ZOOKEEPER_QUORUMS')
-    kafka_brokers = os.getenv('KAFKA_BROKERS')
+def add_dns_domain(hosts):
+    def add_dns_domain_to_host(host):
+        port = None
+        if ':' in host:
+            host, port = host.split(':', 1)
+        if '.' not in host:
+            host += DNS_DOMAIN
+        return host + ':' + port if port else host
+    return ','.join(map(add_dns_domain_to_host, hosts.split(',')))
 
-    if zk_quorums is None:
-        raise RuntimeError('MONITORED_ZOOKEEPER_QUORUMS not found')
 
-    if kafka_brokers is None:
-        raise RuntimeError('KAFKA_BROKERS not found')
+def parse_kafka_info(kafka):
+    # 1=2,3=4 --> {1:2, 3:4}
+    info = dict(part.split('=') for part in kafka.split(' '))
+    if 'env' not in info or 'kafka' not in info or 'zk' not in info:
+        raise RuntimeException('Kafka info must contain "env", "kafka", and "zk"')
+    zk = KazooClient(hosts=add_dns_domain(info['zk']))
+    kafka = KafkaClient(add_dns_domain(info['kafka']))
+    env = info['env']
+    return env, kafka, zk
 
-    # we could just use ZK to find all topics, should we just do that?
-    topics = ['mobile_metrics', 'raw_events']
+def report():
+    # see the monitoring.json chef role
+    kafkas = os.getenv('KAFKA_BROKERS')
 
-    consumer_group = 'secor_group'
+    if kafkas is None:
+        raise RuntimeError('KAFKA_BROKERS not found')
 
-    for zk_quorum in zk_quorums.split('|'):
-        zk = KazooClient(hosts=zk_quorum)
-        kafka = KafkaClient(kafka_brokers)
+    consumer_groups = ['secor_group']
 
+    for kafka in kafkas.split('|'):
+        env, kafka, zk = parse_kafka_info(kafka)
         zk.start()
+        topics = get_kafka_topics(zk)
         for topic in topics:
-            total_broker_offset = report_broker_info(kafka, zk, topic)
-            total_consumer_offset = 0
-            for partition in get_partitions(zk, consumer_group, topic):
-                offset = get_consumer_group_offset(zk, consumer_group, topic, partition)
-                total_consumer_offset += offset
-                print format_kafka_consumer_offset_tsd_key(consumer_group, topic, partition, offset)
-            print format_kafka_consumer_lag_tsd_key(consumer_group, topic, total_broker_offset - total_consumer_offset)
+            total_broker_offset = report_broker_info(env, kafka, zk, topic)
+            for consumer_group in consumer_groups:
+                total_consumer_offset = 0
+                for partition in get_partitions(zk, topic):
+                    offset = get_consumer_group_offset(zk, consumer_group, topic, partition)
```

Comment:
that seems sort of excessive at this point - it's possible to do that and i agree higher granularity == better, but it just seems like overkill